src/Agent.Listener/CommandSettings.cs

```diff
@@ -127,6 +127,39 @@ public bool GetRunAsService()
                 defaultValue: false);
         }
 
+        public bool GetEnableAutoLogon()
+        {
+            return TestFlagOrPrompt(
+                name: Constants.Agent.CommandLine.Flags.EnableAutoLogon,
+                description: StringUtil.Loc("EnableAutoLogon"),
+                defaultValue: !Unattended);
```

Comment:
default value depending on condition is not a good idea. it should be either false/true and not to depend